cmd/dnslol/main.go

```diff
@@ -211,6 +204,10 @@ func main() {
 		names <- name
 	}
 
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
```

Comment:
Did you mean to change the old `log.Fatalf` with a prefixed message into a raw `log.Fatal(err)`? I think I prefer adding the additional context since this is where the buck stops w.r.t displaying the information to a user. The err isn't passed back up to a higher layer that renders it with context.

```suggestion
		log.Fatalf("Error reading names from standard in: %v\n", err)
```